MaxText/max_utils.py

```diff
@@ -180,46 +180,74 @@ def maybe_initialize_jax_distributed_system(raw_keys):
     max_logging.log("Attempting to initialize the jax distributed system...")
     jax.distributed.initialize()
     max_logging.log("Jax distributed system initialized!")
+  elif is_gpu_backend(raw_keys):
+    max_logging.log("Attempting to initialize the jax distributed system for GPU backend...")
+    initialize_jax_for_gpu(raw_keys)
+    max_logging.log("Jax distributed system initialized on GPU!")
   elif is_cpu_backend(raw_keys):
     max_logging.log("Attempting to initialize the jax distributed system for CPU backend...")
     initialize_jax_for_cpu()
     max_logging.log("Jax distributed system initialized on CPUs!")
 
+def initialize_jax_for_gpu(raw_keys):
+  """Jax distributed initialize for GPUs. Includes retries until the coordinator is ready.
+  """
+  coordinator_ip_address = get_coordinator_ip_address()
+  coordinator_port = str(os.getenv("JAX_COORDINATOR_PORT"))
+
+  jax.distributed.initialize(
+      coordinator_address=f"{coordinator_ip_address}:{coordinator_port}",
+      num_processes=int(os.getenv("JAX_NUM_PROCESSES")),
+      process_id=raw_keys["process_id"],
```

Comment:
isn't this supposed to come from an env var?

Reply:
Done
1. Resolving the IP and setting it as environment variable
2. Consuming the Coordinator IP envvar